apps/api/v2/src/modules/organizations/services/event-types/input.service.ts

```diff
@@ -19,37 +19,90 @@ export class InputOrganizationsEventTypesService {
     private readonly usersRepository: UsersRepository,
     private readonly orgEventTypesRepository: OrganizationsEventTypesRepository
   ) {}
+  async transformAndValidateCreateTeamEventTypeInput(
+    userId: number,
+    teamId: number,
+    inputEventType: CreateTeamEventTypeInput_2024_06_14
+  ) {
+    await this.validateHosts(teamId, inputEventType.hosts);
+
+    const transformedBody = await this.transformInputCreateTeamEventType(teamId, inputEventType);
+
+    await this.inputEventTypesService.validateEventTypeInputs(
+      undefined,
```

Comment:
Seems odd to pass undefined to a function. `validateEventTypeInputs` has 5 parameters and all of them are optional, which makes it very hard to understand which ones should be passed and how do their relate to each other any why.

If possible need to have multiple smaller functions that have all parameters required so that we can understand how do they interrelate.

Reply:
https://github.com/calcom/cal.com/commit/02b4d70831c573ccacfc8f8778e9fecf26b2e123